### PR TITLE
[SPARK-46492][BUILD] Simplify the Java version check in `SparkBuild.scala`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to simplify the Java version check in `SparkBuild.scala` due to Apache Spark 4.0 minimum supports Java 17

### Why are the changes needed?
Clean up no longer needed Java version checks.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -294,21 +294,16 @@ object SparkBuild extends PomBuild {
 
     javaOptions ++= {
       val versionParts = System.getProperty("java.version").split("[+.\\-]+", 3)
-      var major = versionParts(0).toInt
+      val major = versionParts(0).toInt
       if (major >= 21) {
         Seq("--add-modules=jdk.incubator.vector", "-Dforeign.restricted=warn")
-      } else if (major >= 16) {
```
Comment (LuciferYang) [inline project/SparkBuild.scala:300]: always >= 17 now

File: project/SparkBuild.scala [hunk 2]
```diff
@@ -294,21 +294,16 @@ object SparkBuild extends PomBuild {
...
-        Seq.empty
+        Seq("--add-modules=jdk.incubator.vector,jdk.incubator.foreign", "-Dforeign.restricted=warn")
       }
     },
 
     (Compile / doc / javacOptions) ++= {
-      val versionParts = System.getProperty("java.version").split("[+.\\-]+", 3)
-      var major = versionParts(0).toInt
-      if (major == 1) major = versionParts(1).toInt
-      if (major >= 8) Seq("-Xdoclint:all", "-Xdoclint:-missing") else Seq.empty
```
Comment (LuciferYang) [inline project/SparkBuild.scala:311]: always >= 8 now, so return `Seq("-Xdoclint:all", "-Xdoclint:-missing")` as default

File: project/SparkBuild.scala [hunk 3]
```diff
@@ -1427,7 +1418,8 @@ object Unidoc {
         "-tag", "constructor:X",
         "-tag", "todo:X",
         "-tag", "groupname:X",
-      ) ++ { if (major >= 9) Seq("--ignore-source-errors", "-notree") else Seq.empty }
```
Comment (LuciferYang) [inline project/SparkBuild.scala:1430]: always >= 8 now, so append `"--ignore-source-errors", "-notree"` as default

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Pull request closed without merging**